softlayer/resource_softlayer_user.go

```diff
@@ -474,6 +474,13 @@ func resourceSoftLayerUserExists(d *schema.ResourceData, meta interface{}) (bool
 
 	result, err := service.Id(id).GetObject()
 
+	// When a user is deleted, it has remained with "CANCEL_PENDING" status in specific time period.
+	// The user with this status should be considered as a non-exist user.
+	if result.UserStatus != nil && result.UserStatus.KeyName != nil &&
+		*result.UserStatus.KeyName == "CANCEL_PENDING" {
+		return false, nil
+	}
+
```

Comment:
Added codes to detect 404 error and other errors:
404 error -> return false, nil
Other errors(Such as authentication problem or network error) -> return false, Error